### PR TITLE
Remove ipoverusb connection from build window, add link to usbncm docs

### PR DESCRIPTION
**Pull request by davidkline-ms**
This change removes the use of ip-over-usb connections in the build and deploy window for UWP projects. The recommended mechanism for connecting to the HoloLens via USB is to use usb-ncm.

To help developer identify the correct IP address to use for the usb-ncm connection ("169.254.x.x") a button labeled "Connect over USB" was added to link to the documentation on how to identify the appropriate address.

image

- Fixes: #10322

### CODE REVIEW

File: Assets/MRTK/Tools/BuildWindow/BuildDeployWindow.cs [hunk 1]
```diff
@@ -888,13 +823,15 @@ private void RenderRemoteConnections()
...
                 }
 
-                if (IsHoloLensConnectedUsb && IsLocalConnection(currentConnection))
+                using (new EditorGUILayout.HorizontalScope())
                 {
-                    EditorGUILayout.LabelField(HOLOLENS_USB);
+                    currentConnection.IP = EditorGUILayout.TextField(IPAddressLabel, currentConnection.IP, GUILayout.Width(HALF_WIDTH));
+                    if (GUILayout.Button("Connect over USB", GUILayout.Width(128f)))
+                    {
+                        Application.OpenURL("https://docs.microsoft.com/en-us/windows/mixed-reality/develop/advanced-concepts/using-the-windows-device-portal#connecting-over-usb");
```
Comment (MaxWang-MS) [inline Assets/MRTK/Tools/BuildWindow/BuildDeployWindow.cs:831]: Could you please remove the locale ("en-us") in the URL?

File: Assets/MRTK/Tools/BuildWindow/BuildDeployWindow.cs [hunk 2]
```diff
@@ -1502,7 +1400,7 @@ private static async Task ExecuteActionAsync(Func<DeviceInfo, Task> exec)
 
             if (UwpBuildDeployPreferences.TargetAllConnections)
             {
-                targetDevices = new List<DeviceInfo>() { localConnection };
+                targetDevices = new List<DeviceInfo>(); ;
```
Comment (MaxWang-MS) [inline Assets/MRTK/Tools/BuildWindow/BuildDeployWindow.cs:1403]: Extra ;

**Comment by keveleigh** [comment]
@davidkline-ms you'll likely want to merge with https://github.com/microsoft/MixedRealityToolkit-Unity/pull/10571 to test successful deployment. Maybe your PR will fix the deployment issue I'm seeing! My PR fixes the Rest calls not completing, though.

File: Assets/MRTK/Tools/BuildWindow/BuildDeployWindow.cs [hunk 3]
```diff
@@ -252,16 +248,6 @@ private static int CurrentRemoteConnectionIndex
             }
         }
 
-        /// <summary>
-        /// Tracks whether the current UI preference is to target the local machine or remote machine for deployment.
-        /// Saves state for duration of current Unity session
-        /// </summary>
-        private static bool UseRemoteTarget
-        {
-            get => SessionState.GetBool(UseRemoteTargetSessionKey, false);
```
[the remark this answers is not in the document]
Comment (davidkline-ms) [reply Assets/MRTK/Tools/BuildWindow/BuildDeployWindow.cs:261]: yes! :smile:

**Comment by davidkline-ms** [comment]
closing in favor of #10575

**Pull request closed without merging**